### PR TITLE
Release Version 2.9

### PR DESCRIPTION
**Pull request by kprinssu**
Notable changes:

- [AMD support](https://github.com/VoodooI2C/VoodooI2C/commit/f9f703b760711e25bd094058ecb6f19dea52dc5f) within VoodooI2C (thanks to @ChefKissInc)
-  VoodooGPIO has [AMD support](https://github.com/VoodooI2C/VoodooGPIO/commit/e0ca3d2e08462ff940b97eaef3271c69d8da6db3) (thanks @ChefKissInc and @qhuyduong)
-  [Physical button fixes](https://github.com/VoodooI2C/VoodooI2CHID/commit/9ebbc8f823e54924dc24a43000a61c3e34cb6800) for VoodooI2CHID (thanks @1Revenger1)
- [Improvements to the touchscreen driver](https://github.com/VoodooI2C/VoodooI2CHID/commit/858c6f229e91c4ebca67883575ffb7d005305fe2) for VoodooI2CHID (thanks @lazd)
- [Better for support for Chromebooks](https://github.com/VoodooI2C/VoodooGPIO/pull/15#event-9642129243) (thanks @1Revenger1)
- Palm rejection for VoodooI2C, VoodooI2CHID and VoodooI2CELAN (thanks @1Revenger1 )

This PR contains various improvements. Huge shout to @ChefKissInc  for AMD support, @1Revenger1 for continuous improvements, and @lazd for improving touchscreen support for Surface devices. 

Thanks @newperson1746  and @1Revenger1  for the improved Github Actions and fixes.

### CODE REVIEW

**Comment by lazd** [comment]
I think it would be wise to get https://github.com/VoodooI2C/VoodooI2CHID/pull/69 in before releasing 2.9 as it contains some fixes for my previous PR mentioned in the description above. There are open questions about pens, but I think the touchscreen interaction fixes should definitely go in and we can see if pens work via feedback from users.

**Comment by kprinssu** [comment]
Thanks for the callout @lazd. This release will be delayed until later this year.

**Comment by denyx01** [comment]
On my Synapatics track pad, the most updated version 2.8 is not working, so doing a test replacing files I noticed after placing the current folder "MacOS" the track pad stopped, therefore keeping the VoodooIC2 2.7.1 version kept working. I have problem in sleep no, where when entering safely the track pad to work.

**Comment by kprinssu** [comment]
@denyx01 This pull request is not for support.

**Comment by hacker1024** [comment]
The aforementioned PR has now been merged.

**Comment by SiddharthManthan** [comment]
Please release this version as early as possible. Many AMD laptops are lacking touchpad support.

**Comment by VisualEhrmanntraut** [comment]
@SiddharthManthan I've got a build in https://chefkissinc.github.io/guide/gathering-files/kexts#input. It's not 100% up-to-date as the guide is currently undergoing a complete rewrite, but it should work for now.

**Comment by kprinssu** [comment]
> Please release this version as early as possible. Many AMD laptops are lacking touchpad support.

This is a PR to track changes in the new version. This is not a place to spam asking it for released.

**Comment by stevezhengshiqi** [comment]
Hi @kprinssu I am sorry if this bothers you. Build from the latest commit https://github.com/VoodooI2C/VoodooI2C/pull/530 will freeze trackpad on my ELAN.
As I perceive so far, https://github.com/VoodooI2C/VoodooGPIO/commit/e0ca3d2e08462ff940b97eaef3271c69d8da6db3 is in good shape, and I am happy to see it pushed and watch AMD users' feedback.

**Comment by kprinssu** [comment]
> Hi @kprinssu I am sorry if this bothers you. Build from the latest commit #530 will freeze trackpad on my ELAN. As I perceive so far, [VoodooI2C/VoodooGPIO@e0ca3d2](https://github.com/VoodooI2C/VoodooGPIO/commit/e0ca3d2e08462ff940b97eaef3271c69d8da6db3) is in good shape, and I am happy to see it pushed and watch AMD users' feedback.

That commit is pretty big and introduces AMD support. I am unsure what could cause breakage with your device.

**Comment by LukeMech** [comment]
> [Built.zip](https://github.com/VoodooI2C/VoodooI2C/files/14827413/Built.zip) Here is a build of I2C and satellites for 2.9. Testing and feedback could be helpful. (target is set to 14.4)

Build doesn't include VoodooI2CHID "recent" pushes I think, or they don't help in Lenovo Yoga 3rd Pen (it works up to 2.6.5 version, I tried compiling myself with latest VoodooI2CHID repo commits but ran into some problems)

**Comment by 1Revenger1** [comment]
If you don't mind testing out palm rejection and only have Elan/HID devices, this [build](https://github.com/VoodooI2C/VoodooI2C/pull/547#issuecomment-2076129321) here should work.

**Comment by LukeMech** [comment]
I was sure I've got Synaptics touchpad, but deleting Synaptics kext didn't break it and using just mentioned by you files did a trick - pen works again, thanks!

**Comment by LeeBinder** [comment]
@LukeMech download, unzip and run [DIsplay info about ELAN Trackpad.zip](https://github.com/VoodooI2C/VoodooI2C/files/15444588/DIsplay.info.about.ELAN.Trackpad.zip)

Content:
```

#!/bin/bash
clear
ioreg -l | grep ELAN
echo
# source: https://github.com/Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh/issues/12#issuecomment-1332945899
ioreg -rxn IOHIDInterface -k "VoodooI2CServices Supported" -t | grep -E 'IOACPIPlatformDevice|IOPCIDevice|VoodooI2CDeviceNub|VendorID|ProductID|HIDEventDriver  <class VoodooI2C|Interface  <class VoodooI2C' | cut -d "<" -f1
```

**Comment by LukeMech** [comment]
> @LukeMech download, unzip and run [DIsplay info about ELAN Trackpad.zip](https://github.com/VoodooI2C/VoodooI2C/files/15444588/DIsplay.info.about.ELAN.Trackpad.zip)
> 
> Content:
> 
> ```
> 
> #!/bin/bash
> clear
> ioreg -l | grep ELAN
> echo
> # source: https://github.com/Qonfused/ASUS-ZenBook-Duo-14-UX481-Hackintosh/issues/12#issuecomment-1332945899
> ioreg -rxn IOHIDInterface -k "VoodooI2CServices Supported" -t | grep -E 'IOACPIPlatformDevice|IOPCIDevice|VoodooI2CDeviceNub|VendorID|ProductID|HIDEventDriver  <class VoodooI2C|Interface  <class VoodooI2C' | cut -d "<" -f1
> ```

Nothing reported :p
Zrzut ekranu 2024-05-25 o 20 48 03

**Comment by LeeBinder** [comment]
run again and scroll up (looks as if your scrolled down too much)

**Comment by LukeMech** [comment]
> run again and scroll up (looks as if your scrolled down too much)

Nah, there is just nothing (I'm sure it's synaptics, it's lenovo x1 yoga 3rd laptop)
Zrzut ekranu 2024-05-25 o 20 58 47

**Comment by LeeBinder** [comment]
Very most likely. What's above all important is that your yoga's pen works again 👍

**Comment by stevezhengshiqi** [comment]
@kprinssu Hi, I see the pip install cldoc command does not go through due to the error Python environment. I suggest adding the following stage before `src=...` command in the yaml file.
```
      # Python 3.10 is not compatible with VoodooI2C cldoc
      - uses: actions/setup-python@v5
        with:
            python-version: '3.9'
      - run: src=$(/usr/bin/curl -Lfs https://raw.githubusercontent.com/acidanthera/VoodooInput/master/VoodooInput/Scripts/bootstrap.sh) && eval "$src" && mv VoodooInput Dependencies
      - run: pip3 install -q cpplint || exit 1
      - run: pip3 install -q git+https://github.com/VoodooI2C/cldoc.git || exit 1
      - run: git submodule init && git submodule update
...
```
which works in my GitHub Action setup.

File: .github/workflows/objective-c-xcode.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: download mackernelsdk
+        run: git clone https://github.com/acidanthera/MacKernelSDK
+
+      - run: src=$(/usr/bin/curl -Lfs https://raw.githubusercontent.com/acidanthera/VoodooInput/master/VoodooInput/Scripts/bootstrap.sh) && eval "$src" && mv VoodooInput Dependencies
+      - run: pip3 install --break-system-packages cpplint
+      - run: pip3 install --break-system-packages git+https://github.com/newperson1746/cldoc-fix.git
```
Comment (kprinssu) [inline .github/workflows/objective-c-xcode.yml:24]: This is a security concern and we should change this to pull the cldoc package from the VoodooI2C repo.

**Review by 1Revenger1 — APPROVED** [review]
Tested out the build artifacts on my Chromebook - LGTM

**Pull request merged**